math/src/field/fields/u64_prime_field.rs

```diff
@@ -249,4 +260,75 @@ mod tests {
         let bytes = vec![1, 0, 0, 0, 0, 0, 0, 0];
         assert_eq!(FE::from_bytes_le(&bytes).unwrap().to_bytes_le(), bytes);
     }
+
+    #[test]
+    fn generate_random_using_rng_generator_and_check_if_normal_distribution() {
+        let mut rng = rand::thread_rng();
+        const LOCAL_MODULUS: u64 = 1663;
+        type FEx = FieldElement<U64PrimeField<LOCAL_MODULUS>>;
+
+        // repeat test n times to observe the statistical behaviour
+        let n: u64 = 100;
+
+        let mut sigma_probs: [f64; 4] = [0.0; 4];
+
+        for _ in 0..n {
+            // Generate a rand distribution of field elements
+            const N_SAMPLES: usize = 200000;
+            let mut rand_dist: [FEx; N_SAMPLES] = [FEx::zero(); N_SAMPLES];
+
+            for i in &mut rand_dist {
+                *i = rng.gen();
+            }
+
+            // Generate histogram where columns are field elements
+            let mut hist: [u64; (LOCAL_MODULUS) as usize] = [0; (LOCAL_MODULUS) as usize];
+
+            for i in 0..=(rand_dist.len() - 1) {
+                hist[*rand_dist[i].value() as usize] += 1;
+            }
+
+            // Compute arithmetic average of histogram columns values
+            let mut avg: u64 = 0;
+
+            for i in hist {
+                avg += i;
+            }
+            avg /= hist.len() as u64;
+
```

Comment:
this could be written as
`let avg = hist.iter().sum() / hist.len() as u64;`

Reply:
Done!